### ISSUE TITLE
Table.add_column speed and reading files > 2 GB

### ISSUE THREAD

**Issue by cdeil** [atpy/atpy#53]
I tried reading a 3GB FITS table and got only zeros.

Then I wrote this script to reproduce the bug:
https://gist.github.com/3796624

I now noticed that the problem is that pyfits can't read files > 2 GB, given by the 32 bit limit, even though my python is 64 bit. Using the most recent astropy.io.fits it works. So I guess this is really a pyfits bug? It should at least warn instead of returning only zeros as data?

What I wanted to mention for atpy is that  repeated Table.add_column calls is a slow way to create a table, because each time a copy of the table data is made. Maybe you can mention this in the docs and suggest a fast way to build a table out of many numpy arrays?

Also note that adding the first column takes 3 sec, but the second column only 0.04 sec.
Even if there is overhead setting up some Table metadata for the first column this shouldn't take 3 seconds, right?

**Comment by cdeil** [comment]
Btw., I did not see a way to write an astropy.table.Table to fits.
I probably overlooked it, or is that really not ready yet?

**Comment by astrofrog** [comment]
@cdeil - could it be related to https://github.com/astropy/astropy/issues/839? Now that Astropy includes the `Table` class, I will not spend time optimizing ATpy. If the Astropy `Table` class has performance issues, please raise it there instead.

**Comment by cdeil** [comment]
I did some tests and indeed it seems that `astropy.table.Table` does better than `atpy.Table`.

Just for reference, here's one more thing I noticed about `atpy.Table`. This creates 4 GB of data and then adds it as a column to an `atpy.Table`:

```
import numpy as np
from atpy import Table
table = Table()
data = np.arange(int(1e9), dtype=np.int32)
table.add_column(name='a', data=data)
```

On my Mac this takes up all free RAM (~12 GB) and starts swapping, although even if a copy is made it should only take 8 GB total.
